solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -755,11 +762,12 @@ private Request fillContentStream(
           content.addFilePart(
               name,
               contentStream.getName(),
-              new InputStreamContentProvider(contentStream.getStream()),
+              new InputStreamRequestContent(contentStream.getStream()),
               fields);
         }
       }
-      req.content(content);
+      content.close();
```

Comment:
Do we have to worry about closing this if there is an exception? not sure if `MultiPartRequestContent` can do try w/ resources?

Reply:
good idea! seems it does so I will give it a go.

Reply:
change is in!